### PR TITLE
refactor/char u/16.1

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/os/env.c [hunk 1]
```diff
@@ -667,7 +667,7 @@ void expand_env_esc(char *restrict srcp, char *restrict dst, int dstlen, bool es
         int c = dstlen - 1;
         while (c-- > 0
                && *tail
-               && vim_isfilec(*tail)
+               && vim_isfilec((uint8_t)(*tail))
```
Comment (zeertzjq) [inline src/nvim/os/env.c:670]: This one is impossible to test on CI as it requires a Unicode user name.

Comment (jamessan) [reply src/nvim/os/env.c:670]: Can't we just add a user as part of CI setup?

Comment (zeertzjq) [reply src/nvim/os/env.c:670]: I thought of this after posting this comment, but it sounds like an overkill.

File: src/nvim/edit.c [hunk 2]
```diff
@@ -2144,35 +2144,35 @@ void insertchar(int c, int flags, int second_indent)
            && MB_BYTE2LEN(c) == 1
            && i < INPUT_BUFLEN
            && (textwidth == 0
-               || (virtcol += byte2cells(buf[i - 1])) < (colnr_T)textwidth)
+               || (virtcol += byte2cells((uint8_t)buf[i - 1])) < (colnr_T)textwidth)
            && !(!no_abbr && !vim_iswordc(c) && vim_iswordc(buf[i - 1]))) {
```
Comment (zeertzjq) [inline src/nvim/edit.c:2148]:
```suggestion
           && !(!no_abbr && !vim_iswordc(c) && vim_iswordc((uint8_t)buf[i - 1]))) {
```

File: src/nvim/edit.c [hunk 3]
```diff
@@ -2144,35 +2144,35 @@ void insertchar(int c, int flags, int second_indent)
...
       c = vgetc();
       if (p_hkmap && KeyTyped) {
         c = hkmap(c);                       // Hebrew mode mapping
       }
-      buf[i++] = (char_u)c;
+      buf[i++] = (char)c;
     }
 
     do_digraph(-1);                     // clear digraphs
     do_digraph(buf[i - 1]);               // may be the start of a digraph
```
Comment (zeertzjq) [inline src/nvim/edit.c:2157]:
```suggestion
    do_digraph((uint8_t)buf[i - 1]);               // may be the start of a digraph
```

File: src/nvim/edit.c [hunk 4]
```diff
@@ -2144,35 +2144,35 @@ void insertchar(int c, int flags, int second_indent)
...
     buf[i] = NUL;
-    ins_str((char *)buf);
+    ins_str(buf);
     if (flags & INSCHAR_CTRLV) {
       redo_literal(*buf);
```
Comment (zeertzjq) [inline src/nvim/edit.c:2161]:
```suggestion
      redo_literal((uint8_t)(*buf));
```

File: src/nvim/os/env.c [hunk 5]
```diff
@@ -626,15 +626,15 @@ void expand_env_esc(char *restrict srcp, char *restrict dst, int dstlen, bool es
 
 #ifdef UNIX
         // Unix has ${var-name} type environment vars
-        if (*tail == '{' && !vim_isIDc('{')) {
+        if (*tail == '{' && !vim_isIDc((uint8_t)'{')) {
```
Comment (zeertzjq) [inline src/nvim/os/env.c:629]: This is not needed

**Pull request merged**